src/rgw/rgw_rados.h

```diff
@@ -1956,17 +1955,7 @@ class RGWRados
 
   RGWDataChangesLog *data_log;
 
-  virtual ~RGWRados() {
-    for (uint32_t i=0; i < num_rados_handles; i++) {
-      if (rados[i]) {
-        rados[i]->shutdown();
-        delete rados[i];
-      }
-    }
-    if (rados) {
-      delete[] rados;
-    }
-  }
+  virtual ~RGWRados() = default;
```

Comment:
> Does `librados::Rados::~Rados()` do that for us?

```cpp
librados::Rados::~Rados()
{
  shutdown();
}
```

> It looks we're leaking `cr_registry` but this isn't new (no regression here).

correct, `cr_registry` is only freed in `RGWRados::finalize()`. before this patch, it was leaked if the call to `cr_registry->hook_to_admin_command("cr dump");` failed in `RGWRados::init_rados()`. but if `init_rados()` succeeds, it's fair to expect `finalize()` to be called for cleanup